From 5d7f922897e6862ef11731fd0640db696fe7c984 Mon Sep 17 00:00:00 2001
From: Shaofeng Tang <shaofeng.tang@intel.com>
Date: Wed, 12 Oct 2022 11:25:14 +0800
Subject: [PATCH] [WA] force to set dma_mask for virtio_video device

dma_mask must be set in 'probe' for using dma mapping in virtio_video
device.
This issue happen on kernel 5.10 and 5.15.
No issue to use dma mapping on Android R/lts2019-chromium

Tracked-On: OAM-99370
Signed-off-by: Shaofeng Tang <shaofeng.tang@intel.com>
---
 drivers/media/virtio/virtio_video_driver.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/virtio/virtio_video_driver.c b/drivers/media/virtio/virtio_video_driver.c
index 53c466510eaa..7c5dda5c58b7 100644
--- a/drivers/media/virtio/virtio_video_driver.c
+++ b/drivers/media/virtio/virtio_video_driver.c
@@ -193,6 +193,8 @@ static int virtio_video_probe(struct virtio_device *vdev)
 		goto err_res_type;
 	}
 
+	use_dma_mem = 1;
+
 	vv->vdev = vdev;
 	vv->debug = debug;
 	vv->use_dma_mem = use_dma_mem;
@@ -255,6 +257,8 @@ static int virtio_video_probe(struct virtio_device *vdev)
 	if (ret)
 		goto err_events;
 
+	ret = dma_coerce_mask_and_coherent(dev, DMA_BIT_MASK(64));
+
 	virtio_device_ready(vdev);
 	vv->vq_ready = true;
 	vv->got_caps = false;
-- 
2.29.2

